Tutorial.playground/Contents.swift

```diff
@@ -0,0 +1,178 @@
+//: components
+//:
+//: You can connect these components together in four separate ways which
+//: parallel the four above types:
+//:
+//: * `for_` handles `yield`s
+//:
+//: * `>~` handles `await`s
+//:
+//: * `>->` handles both `yield`s and `await`s
+//:
+//: * `>>-` handles return values
+//:
+//: As you connect components their types will change to reflect inputs and
+//: outputs that you`ve fused away.  You know that you`re done connecting things
+//: when you get an `Effect`, meaning that you have handled all inputs and
+//: outputs.  You run this final `Effect` to begin streaming.
+
+import Swiftz
+import Aquifer
+
+//: Producers
+
+//: `Producer`s are effectful streams of input.  Specifically, a `Producer` is a
+//: monad transformer that extends any base monad with a new `yield` command.
+//: This `yield` command lets you send output downstream to an anonymous
+//: handler, decoupling how you generate values from how you consume them.
+
+//: The following `stdinByLine` `Producer` shows how to incrementally read in
+//: `String`s from standard input and `yield` them downstream, terminating
+//: gracefully when reaching the end of the input:
+
+import class Foundation.NSFileHandle
+
+//:                          +----------  A `Producer` that yields `String`s
+//:                          |
+//:                          |       +-- Every monad transformer has a base monad.
+//:                          |       |   This time the base monad is `IO`.
+//:                          |       |
+//:                          |       |     +-- Every action has a return value.
+//:                          |       |     |   This action returns `()` when finished
+//:                          v       v     v
+public func stdinByLine() -> Producer<String, ()>.T {
+	let handle = NSFileHandle.fileHandleWithStandardInput() // Grab the handle
+	if handle.isAtEndOfFile { // If we`re at the end of the line, stop.
+		return pure(())
+	} else {
+		return yield(handle.readLine) >>- { _ in fromHandle(handle) } // Otherwise `yield` the line and loop.
+	}
+}
+
+//: `yield` emits a value, suspending the current `Producer` until the value is
+//: consumed.  If nobody consumes the value (which is possible) then `yield`
+//: never returns.  You can think of `yield` as having the following type:
+//:
+//:     func yield<A>(value : A) -> Producer<A, ()>.T
+//:
+//: The true type of `yield` is actually more general and powerful.  Throughout
+//: the tutorial I will present type signatures like this that are simplified at
+//: first and then later reveal more general versions.  So read the above type
+//: signature as simply saying: "You can use `yield` within a `Producer`, but
+//: you may be able to use `yield` in other contexts, too."
+//:
+//: Click the link to `yield` to navigate to its documentation.  There you will
+//: see that `yield` actually uses the `Producer` (with an apostrophe) type
```

Comment:
This line definitely needs to change to reflect the fact that not only do we not have a `Producer'` type available, but that writing one is actually impossible in Swift, since we have no way to express rank-n polymorphism.